### PR TITLE
Clarify usage of user variables and recipient variables

### PR DESCRIPTION
**Pull request by thrawn01**

### CODE REVIEW

File: source/user_manual.rst [hunk 1]
```diff
@@ -908,8 +910,9 @@ Attaching Data to Messages
 
 When sending, you can attach data to your messages by passing custom data to the API or SMTP endpoints.
 The data will be represented as a header within the email, ``X-Mailgun-Variables``. The data
-is formatted in JSON and included in any webhook events related to the email containing the custom data. Several
-such headers may be included and their values will be combined.
+should be formatted in JSON and will also be included in webhook payloads and events returned from the events API
```
Comment (highlycaffeinated) [inline source/user_manual.rst]: "should" or "must"?

File: source/user_manual.rst [hunk 2]
```diff
@@ -918,14 +921,14 @@ Example::
 
 To add this header to your message:
 
-API: Pass the following parameter, "v:my-custom-data" => "{"my_message_id": 123}".
+HTTP API: Provide a form parameter prefixed with ``v:`` IE: ``v:my_message_id=123``
```
Comment (horkhe) [inline source/user_manual.rst]: I had to google what IE stands for, can you change it to *for example* like in other places of this document.

File: source/user_manual.rst [hunk 3]
```diff
@@ -918,14 +921,14 @@ Example::
 
 To add this header to your message:
```
Comment (horkhe) [inline source/user_manual.rst]: it reads:

> To add this header to your message:
> ...
> SMTP: Add the following header to your email...

So "To add this header ... add the following header..." - very confusing. And also it is not clear what *this* refers to.

**Comment by thrawn01** [comment]
I agree, the entire thing as written is confusing. I re-wrote the entire section to help clarify it's usage.

File: source/user_manual.rst [hunk 4]
```diff
@@ -906,31 +908,48 @@ And here's a sample in Go
...
+For example::
 
-You can also use values from your recipient variables to provide a custom variable per a recipient using
-templating. For example when sending via the API::
+    v:first_name=John
+    v:last_name=Smith
+    v:my_message_id=123
 
-    {'v:Recipient-Id': '%recipient.id%'}
+The data provided will be included the recipients email via a header called ``X-Mailgun-Variables``.
```
Comment (horkhe) [inline source/user_manual.rst]: @thrawn01 are you sure this sentence is correct, I cannot grasp what it means. it may be my bad english though.

**Pull request merged**